File: aurafit-frontend/src/Models/DTOS/OrderResponseDTO.ts
import {Status} from "../Enums/Status.ts";
import {ContactInformation} from "../ContactInformation.ts";
import {OrderItemDTO} from "../OrderItemDTO.ts";


export class OrderResponseDTO {
    id: number;
    totalPrice: number;
    orderItems: OrderItemDTO[];
    orderDate: Date;
    status: Status;
    orderNumber: number;
    contactInformation: ContactInformation;

    constructor(id: number, totalPrice: number, orderItems: OrderItemDTO[], orderDate: Date, status: Status, orderNumber: number, contactInformation: ContactInformation) {
        this.id = id;
        this.totalPrice = totalPrice;
        this.orderItems = orderItems;
        this.orderDate = orderDate;
        this.status = status;
        this.orderNumber = orderNumber;
        this.contactInformation = contactInformation;
    }


}

File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/order/OrderResponseDTO.java
package app.aurafitbackend.order;


import app.aurafitbackend.Beans.ContactInformation;
import app.aurafitbackend.Enums.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderResponseDTO {
    private Long id;
    private BigDecimal totalPrice;
    private List<OrderItemResponseDTO> orderItems;
    private LocalDateTime orderDate;
    private Status status;
    private String orderNumber;
    private ContactInformation contactInformation;
}
